SYCL/SubGroup/helper.hpp

```diff
@@ -146,7 +146,7 @@ template <typename T> void exit_if_not_equal(T *val, T *ref, const char *name) {
 template <> void exit_if_not_equal(half val, half ref, const char *name) {
   int16_t cmp_val = reinterpret_cast<int16_t &>(val);
   int16_t cmp_ref = reinterpret_cast<int16_t &>(ref);
-  if (std::abs(cmp_val - cmp_ref) > 1) {
+  if (std::abs(cmp_val - cmp_ref) > 2) {
```

Comment:
Do you know why we need this at all? Other fp types seem to be checked for exact equality (line 112).

Reply:
This compares the results on the host vs device. The host does not have support for `_Float16` so this is implemented with `uint16_t` data type and converting into 32-bit float for arithmetic and back to `uint16_t`.
The device has support for `_Float16`, so it is performing the same operations at 16-bit precision. So the difference in the precision/computation between host and device means there is a larger difference in output, compared to other data types.